base/compiler/ssair/irinterp.jl

```diff
@@ -14,7 +14,7 @@ function concrete_eval_invoke(interp::AbstractInterpreter,
     argtypes = collect_argtypes(interp, inst.args[2:end], nothing, irsv)
     argtypes === nothing && return Pair{Any,Bool}(Bottom, false)
     effects = decode_effects(code.ipo_purity_bits)
-    if is_foldable(effects) && is_all_const_arg(argtypes, #=start=#1)
+    if (is_foldable(effects) && is_all_const_arg(argtypes, #=start=#1) && is_nonoverlayed(mi.def::Method))
```

Comment:
This doesn't seem properly recursive in the `!isoverlayed` check, without a check for `effects.nonoverlayed`, but I don't really know this code very well.

Reply:
Ah, good point. We need a check for `effects.nonoverlayed` here too.